src/bundle/Controller/LocationController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * File containing the LocationController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUiBundle\Controller;
+
+use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use EzSystems\HybridPlatformUi\Form\UiFormFactory;
+use EzSystems\HybridPlatformUi\Repository\UiLocationService;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
+
+class LocationController extends Controller
+{
+    public function actionsAction(
+        Content $content,
+        Request $request,
+        UiLocationService $uiLocationService,
+        RouterInterface $router,
+        UiFormFactory $formFactory
+    ) {
+        $actionsForm = $formFactory->createLocationsActionForm();
+        $actionsForm->handleRequest($request);
+
+        if ($actionsForm->isValid()) {
+            $locationIds = array_keys($actionsForm->get('removeLocations')->getData());
+
+            if ($actionsForm->get('delete')->isClicked()) {
+                $uiLocationService->deleteLocations($locationIds, $content->id);
+            }
+        }
+
+        return new RedirectResponse(
+            $router->generate(
+                '_ez_content_view',
+                [
+                    'contentId' => $content->id,
```

Comment:
that's an interesting topic :) and indeed with the tab + async system we have to find a solution. Since we are at the very beginning of the tab + async implementation (and not about to release a stable version), I agree with @bdunogier, we can avoid that complicated question by not allowing to remove the currently viewed Location **for now**. Of course we need a new story for that problem (explaining the topic and pointing to that thread).
I have some ideas to overcome that issue but I need to think a bit about it